app/controllers/api/visioplainte/creneaux_controller.rb

```diff
@@ -1,41 +1,75 @@
 class Api::Visioplainte::CreneauxController < Api::Visioplainte::BaseController
+  before_action :validate_date_debut_and_service
   before_action :validate_date_range, only: [:index]
+
   def index
-    motif = find_motif
+    creneaux = Users::CreneauxSearch.new(
+      lieu: nil,
+      user: nil,
+      motif: motif,
+      date_range: date_range
+    ).creneaux
 
     render json: {
-      creneaux: creneaux(motif).map do |creneau|
-        {
-          starts_at: creneau.starts_at.iso8601,
-          duration_in_min: motif.default_duration_in_min,
-        }
+      creneaux: creneaux.map do |creneau|
+        creneau_to_hash(creneau)
       end,
     }
   end
 
   def prochain
-    # Une fausse implémentation pour la documentation
-    if params[:service] == "Gendarmerie"
+    date_debut = Date.parse(params[:date_debut])
+    date_range = date_debut..(date_debut + 60.days)
+
+    next_availability = Users::CreneauxSearch.new(
+      lieu: nil,
+      user: nil,
+      motif: motif,
+      date_range: date_range
+    ).next_availability
+
+    if next_availability
+      render json: creneau_to_hash(next_availability)
+    else
       errors = ["Aucun créneau n'est disponible après cette date pour ce service."]
 
       render(json: { errors: errors }, status: :not_found) and return
     end
-
-    render json: {
-      starts_at: "2024-12-22T10:00:00+02:00",
-      duration_in_min: 30,
-    }
   end
 
   private
 
-  def validate_date_range
+  def motif
+    @motif ||= Motif.joins(organisation: :territory).where(territories: { name: Territory::VISIOPLAINTE_NAME })
+      .joins(:service).find_by(service: { name: service_names[params["service"]] })
```

Comment:
Purement cosmétique, mais en général on accès aux param avec des symboles. :wink: 

```suggestion
      .joins(:service).find_by(service: { name: service_names[params[:service]] })
```